### PR TITLE
Add helper to get hosts on metricbeat integration tests

### PR DESCRIPTION
**Pull request by jsoriano**
Add a helper to automatically obtain the host of a service in metricbeat
system tests so tests don't depend on environment variables or
current running docker services. This is needed to allow in the future
to run multiple versions of the same service, is also helpful when
running tests locally.

Environment variables with the format `SERVICE_HOST`, being SERVICE
the uppercased name of the compose service can be used to override the
automatic value. By now `OVERRIDE_HOST` needs to be set to for these
variables to work in the python-based tests.

`NO_COMPOSE` environment variable can be used now to avoid stopping
the running containers too.

Add ports to all services in docker compose files, required for the
helper.

Part of elastic/beats#7957 for python-based integration tests.

### CODE REVIEW

**Comment by jsoriano** [comment]
jenkins, test this again please

File: metricbeat/module/kafka/_meta/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+ARG KAFKA_VERSION=2.1.0
 
 ENV KAFKA_HOME /kafka
-# The advertised host is kafka. This means it will not work if container is started locally and connected from localhost to it
 ENV KAFKA_LOGS_DIR="/kafka-logs"
 ENV _JAVA_OPTIONS "-Djava.net.preferIPv4Stack=true"
 ENV TERM=linux
 
-RUN apt-get update && apt-get install -y curl openjdk-8-jre-headless netcat dnsutils
+RUN apk add -u bash
```
Comment (exekias) [inline metricbeat/module/kafka/_meta/Dockerfile]: changes to this file and `run.sh` probably belong to a different PR?

File: metricbeat/tests/system/test_jolokia.py [hunk 2]
```diff
@@ -50,5 +50,4 @@ def test_jmx(self, mbean):
         assert evt["jolokia"]["test"]["gc"]["collection_count"] >= 0
 
     def get_hosts(self):
-        return [os.getenv('JOLOKIA_HOST', 'localhost') + ':' +
-                os.getenv('JOLOKIA_PORT', '8778')]
+        return [self.compose_host()]
```
Comment (exekias) [inline metricbeat/tests/system/test_jolokia.py]: I see this pattern a lot, you could also replace `get_hosts()` call with `self.compose_host()` and remove `get_hosts` all together??

File: metricbeat/module/kafka/_meta/Dockerfile (hunk 1 above)
Comment (jsoriano) [reply metricbeat/module/kafka/_meta/Dockerfile]: Changes here could probably be reverted, yes, I think they only reduced the size of the image.

Changes in run.sh are needed, kafka announces the addresses clients should use, current container only works if tests can connect to "kafka:9092". After this change services can be listening on random addresses and ports, so we need some plumbing to let kafka know what address it should announce.
But said that, not all changes are needed now, I got all changes done in #7957, that are intended also to be able to have multiple versions of the same service running at the same time, this is not needed till I merge the Go-part of this PR.

I will reduce the changes in these files to the ones really needed for these changes.

File: metricbeat/tests/system/test_jolokia.py (hunk 2 above)
Comment (jsoriano) [reply metricbeat/tests/system/test_jolokia.py]: Good suggestion, thanks! It should be possible, I will do it and override only when needed.

**Comment by jsoriano** [comment]
> I wonder if we can detect ports as exposed in the `Dockerfile`, without explicitly exposing them in `docker-compose.yml`, or maybe that's too hidden

I tried, but I didn't manage to do it, it seems that docker compose only expose ports defined in the docker-compose.yml file. If there is a way to make it expose the ports defined in the Dockerfile we could consider doing it, yes.
In any case it is in my plans to move all the testing code of each module to the module itself, including the docker compose definitions. In many cases we are only using the Dockerfile to define the healthcheck and this can be done in the docker-compose.yml now, so many Dockerfiles might disappear in the future.

**Comment by jsoriano** [comment]
I have seen Kafka tests to fail a couple of times in travis, I am taking a look to this.

**Comment by jsoriano** [comment]
Ok, it fails because it is not ready yet to have multiple docker compose files in the same run, I will try to refactor Kafka docker files to don't need to bring here more changes from #7957

I don't know though why it doesn't fail on Jenkins.

**Comment by jsoriano** [comment]
Issues fixed by reverting more uneeded changes for kafka, this would be ready for another review.
